src/librustc/traits/query/dropck_outlives.rs

```diff
@@ -208,15 +207,15 @@ pub fn trivial_dropck_outlives<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>) -> bool {
         | ty::Error => true,
 
         // [T; N] and [T] have same properties as T.
-        ty::Array(ty, _) | ty::Slice(ty) => tcx.trivial_dropck_outlives(ty),
+        ty::Array(ty, _) | ty::Slice(ty) => trivial_dropck_outlives(tcx, ty),
```

Comment:
Maybe add some comment on why this can't result in infinite recursion? Since the query system isn't around to prevent that. This function could perhaps be a method on `TyCtxt` to avoid changing use sites.